static/js/components/EmploymentTab.js

```diff
@@ -65,56 +79,232 @@ class EmploymentTab extends ProfileTab {
     );
   }
 
-  employmentEntriesForm (index) {
-    let keySet = (key) => ['work_history', index, key];
+  componentWillMount () {
+    const { ui, setWorkHistoryEdit } = this.props;
+    if ( _.isUndefined(ui.workHistoryEdit) ) {
+      setWorkHistoryEdit(true);
+    }
+  }
+
+  toggleWorkHistoryEdit = () => {
+    const { ui, setWorkHistoryEdit } = this.props;
+    setWorkHistoryEdit(!ui.workHistoryEdit);
+  }
+
+  closeWorkDialog = () => {
+    const { setWorkDialogVisibility } = this.props;
+    setWorkDialogVisibility(false);
+  }
+
+  addWorkHistoryEntry = () => {
+    const {
+      updateProfile,
+      profile,
+      setWorkDialogIndex,
+      setWorkDialogVisibility,
+    } = this.props;
+    let clone = Object.assign({}, profile);
+    clone['work_history'] = clone['work_history'].concat(
+      _.cloneDeep(this.blankWorkHistoryEntry)
```

Comment:
I changed it to this:

```js
    let clone = _.cloneDeep(profile);
    clone['work_history'] = clone['work_history'].concat(this.blankWorkHistoryEntry);
```

without cloning profile it's possible to achieve a situation where two of the work_history array entries point at the same object, so that editing one of them appears to edit both entries in the UI.

![prototypes](https://cloud.githubusercontent.com/assets/6207644/15300705/02ab95aa-1b78-11e6-8074-6f30d3c19c8d.png)

You're right that we don't need to clone `this.blankWorkHistoryEntry` though. The above was with this code:

```js
let clone = Object.assign({}, profile, {
    'work_history': profile.work_history.concat(this.blankWorkHistoryEntry)
});
```

steps to repro it:

1. remove all entries (click the X)
2. add two entries without saving either (hit the cancel button when the dialog opens)
3. edit one, see edits replayed twice (two entries in the array are pointing at the same object).